docs/user_guide/high_availabilty.rst

```diff
@@ -0,0 +1,200 @@
+
+If I’m currently hot, and the hot SP has changed to not me, then I transition to the Hot-to-Cold state. In this state,
+I will prepare to stop serving the client requests. If any requests are received during the Hot-to-Cold state, I will
+tell them I am not in service. This is a transition state to the cold state.
+
+Admin Client
+------------
+No response from Overseer (connection error, etc.)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+If I already have a hot SP, I’ll keep using it. I’ll keep retrying to communicate with the Overseer to obtain a response.
+
+No hot SP available
+^^^^^^^^^^^^^^^^^^^
+If I already have a hot SP, I’ll keep using it. I’ll keep retrying to communicate with the Overseer to obtain a
+different response.
+
+Hot SP has not changed
+^^^^^^^^^^^^^^^^^^^^^^
+I’ll continue to use the current hot FL server.
+
+Hot SP has changed
+^^^^^^^^^^^^^^^^^^
+I’ll try to login to the new hot FL server. After that, I will issue commands to the new hot server. If I run
+into any communication issues with the new server, I will keep retrying until success or the hot server
+endpoint changes again.
+
+**************************
+Job Execution Continuation
+**************************
+The secondary feature of HA is the continuation of job execution after SP cutover. NVIDIA FLARE implements a
+snapshot-based job continuation mechanism.
+
+    - Once a job is started, the server creates the first snapshot that remembers the basic job state (job ID,
+      workspace, etc.).
+    - During the execution of the job, the Controller initiates the creation of additional snapshots, based on its
+      own control logic. Some controllers may decide not to create additional snapshots. For example, the
+      Scatter-and-Gather controller works based on the concept of rounds, and it creates a snapshot after each round;
+      whereas the cross-site-validation controller doesn’t create any snapshots.
+    - After the SP cutover, the Controller will initiate the restoration of job execution state from the latest snapshot.
+      If the Controller didn’t create additional snapshots, then the job will be executed from the beginning after the SP cutover.
+    - After the job execution completes, the job snapshot will be deleted from the snapshot storage. If the SP cutover
+      occurs after the job execution completes, the completed job will not be migrated over.
+
+************************
+HA Running Job Migration
+************************
+All the FLComponents in the FL workflow have the option to implement the StatePersistable, which is to decide what
+kind of data needs to persist and migrate to another server in the case of HA SP cutover. The FL snapshot includes
+the current running state of all the FLComponents, the FLContext, and the current Job workspace. Once the HA SP cutover
+occurs, the new SP will restore the FLContext, the Job workspace, and all the components' working states.
```

Comment:
Do we need to explain when exactly the state is persisted and get restored?
Like the end of each round or something like that?
So still a portion of work might be lost.

@yhwen can add more details to this as well.